server/jetstream_cluster_3_test.go

```diff
@@ -2779,3 +2779,47 @@ func TestJetStreamClusterInterestPolicyEphemeral(t *testing.T) {
 		})
 	}
 }
+
+// TestJetStreamClusterWALBuildupOnNoOpPull tests whether or not the consumer
+// RAFT log is being compacted when the stream is idle but we are performing
+// lots of fetches. Otherwise the disk usage just spirals out of control if
+// there are no other state changes to trigger a compaction.
+func TestJetStreamClusterWALBuildupOnNoOpPull(t *testing.T) {
+	c := createJetStreamClusterExplicit(t, "R3S", 3)
+	defer c.shutdown()
+
+	nc, js := jsClientConnect(t, c.randomServer())
+	defer nc.Close()
+
+	_, err := js.AddStream(&nats.StreamConfig{
+		Name:     "TEST",
+		Subjects: []string{"foo"},
+		Replicas: 3,
+	})
+	require_NoError(t, err)
+
+	sub, err := js.PullSubscribe(
+		"foo",
+		"durable",
+		nats.ConsumerReplicas(3),
+	)
+	require_NoError(t, err)
+
+	for i := 0; i < 100000; i++ {
+		_, _ = sub.Fetch(1, nats.MaxWait(time.Microsecond))
+	}
+
+	s := c.randomNonConsumerLeader(globalAccountName, "TEST", "durable")
+	dc := s.js.accounts[globalAccountName].streams["TEST"].consumers["durable"]
+
```

Comment:
Need to wait 5s and then do more fetches then check. Only do a few fetches, like 10.